app/javascript/packages/document-capture/components/address-search.tsx

```diff
@@ -44,21 +45,32 @@ function AddressSearch({
   const ref = useRef<SpinnerButtonRefHandle>(null);
 
   useEffect(() => {
-    if (addressCandidates) {
+    if (addressCandidates && addressCandidates.length > 0) {
+      setNoAddressFoundErrors('');
+      validatedFieldRef.current?.setCustomValidity('');
+      validatedFieldRef.current?.reportValidity();
       const bestMatchedAddress = addressCandidates[0];
       onAddressFound(bestMatchedAddress);
       ref.current?.toggleSpinner(false);
     }
-  }, [addressCandidates]);
+    if (addressCandidates?.length === 0) {
+      setNoAddressFoundErrors('Not a real address');
+      ref.current?.toggleSpinner(false);
+      validatedFieldRef.current?.setCustomValidity(
+        t('in_person_proofing.body.location.inline_error'),
+      );
+      validatedFieldRef.current?.reportValidity();
+    }
```

Comment:
This has a lot of repeated operations that can be simplified, e.g.:
```typescript
const bestMatchedAddress = addressCandidates?.[0];
const addressErrors = bestMatchedAddress ? '' : 'Not a real address';
const validity = bestMatchedAddress ? '' : t('in_person_proofing.body.location.inline_error');

setNoAddressFoundErrors(addressErrors);
ref.current?.toggleSpinner(false);
validatedFieldRef.current?.setCustomValidity(validity);
validatedFieldRef.current?.reportValidity();
bestMatchedAddress && onAddressFound(bestMatchedAddress);
```

Reply:
Great points.  I refactored and would appreciate your thoughts.